scanners/podman_wrapper.py

```diff
@@ -144,17 +181,33 @@ def change_user_id(self, runas_uid, runas_gid):
             raise RuntimeError(f"Unable to retrieve podman UID mapping: {exc}") from exc
 
         # UID mapping
-        self.add_option("--uidmap", f"0:1:{runas_uid}")
-        self.add_option("--uidmap", f"{runas_uid}:0:1")
-        self.add_option(
-            "--uidmap", f"{runas_uid+1}:{runas_uid+1}:{subuid_size-runas_uid}"
-        )
+        if subuid_size >= runas_uid:
+            self.add_option("--uidmap", f"0:1:{runas_uid}")
+            self.add_option("--uidmap", f"{runas_uid}:0:1")
+            self.add_option(
+                "--uidmap", f"{runas_uid+1}:{runas_uid+1}:{subuid_size-runas_uid}"
+            )
+            logging.debug(
+                "podman enabled UID mapping arguments (using uidmap workaround)"
+            )
+        else:
```

Comment:
Shouldn't we raise an Exception in this case? This is an unexpected situation, so the execution shouldn't continue, as we are not setting the mapping

Reply:
Good point. I wasn't sure how to handle various scenario.
If run as root, podman does not need to have subuid enabled, so I just left that as a warning, and podman would fail later on if the user is not root.
So far, I am not aware of rapidast being used as root.
I guess I can raise an exception with a clear explanation and an early fail. If a user requires to use root + podman < 4.3, we can see then.

Reply:
updated accordingly